composables/useNft.ts

```diff
@@ -2,7 +2,7 @@ import type { NFT, NFTMetadata } from '@/components/rmrk/service/scheme'
 import { sanitizeIpfsUrl } from '@/utils/ipfs'
 import { processSingleMetadata } from '@/utils/cachingStrategy'
 import { getMimeType } from '@/utils/gallery/media'
-
+import unionBy from 'lodash/unionBy'
```

Comment:
```ts
const unionBy = (arr, ...args) => {
  let iteratee = args.pop()
  if (typeof iteratee === 'string') {
    const prop = iteratee
    iteratee = item => item[prop]
  }

  return arr
    .concat(...args)
    .filter(
      (x, i, self) => i === self.findIndex(y => iteratee(x) === iteratee(y))
    )
}
```

Reply:
Sorry, but I do not understand why we would write a common util function instead of using lodash. Imo using lodash could avoid some unexpected mistakes.

Reply:
you might not need loadash

Reply:
<img width="921" alt="image" src="https://github.com/kodadot/nft-gallery/assets/31397967/644eb071-8077-43a5-965c-c669ffa1de7a">